pkg/reconciler/managed/changelogger.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package managed
+
+import (
+	"context"
+	"fmt"
+
+	changelogs "github.com/crossplane/crossplane-runtime/apis/changelogs/proto/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// changeLogger processes changes to resources and helps to send them to the
+// change log service
+type changeLogger struct {
+	enabled         bool
```

Comment:
Rather than `enabled bool` I'd suggest:

* Define a ChangeLogger interface
* Have a NopChangeLogger implementation that does nothing when called, and use that by default
* Add a WithChangeLogger option for the Reconciler
* Use WithChangeLogger to pass in this real implementation if the feature flag is enabled

Reply:
This approach of a `ChangeLogger` interface and `GRPCChangeLogger`/`NopChangeLogger` implementations is done in https://github.com/crossplane/crossplane-runtime/pull/754/commits/c23f09b8314e5a7d11e0438243572a0b38cfaff1.